keanu-project/src/test/java/io/improbable/keanu/util/DescriptionGeneratorTest.java

```diff
@@ -0,0 +1,90 @@
+    }
+
+    @Test
+    public void testSimpleDescriptionsCreatedCorrectly() {
+        DoubleVertex two = new ConstantDoubleVertex(2.0);
+        DoubleVertex three = new ConstantDoubleVertex(3.0).setLabel("Three");
+        DoubleVertex four = new ConstantDoubleVertex(4.0).setLabel("Four");
+
+        DoubleVertex result = two.multiply(three).minus(four);
+        String description = descriptionCreator.createDescription(result);
+
+        assertThat(description, is("This Vertex = (Const(2.0) * Three) - Four"));
+    }
+
+    @Test
+    public void testIntegerAdditionAndMultiplicationDescriptionsCreatedCorrectly() {
+        IntegerVertex two = new ConstantIntegerVertex(2);
+        IntegerVertex three = new ConstantIntegerVertex(3).setLabel("Three");
+        IntegerVertex four = new ConstantIntegerVertex(4).setLabel("Four");
+
+        IntegerVertex result = two.multiply(three).plus(four);
+        String description = descriptionCreator.createDescription(result);
+
+        assertThat(description, is("This Vertex = (Const(2) * Three) + Four"));
+    }
+
+    @Test
+    public void testIfVertexDescriptionCreatedCorrectly() {
+        BooleanVertex predicate = new ConstantBooleanVertex(false);
+
+        DoubleVertex three = new ConstantDoubleVertex(3.0).setLabel("Three");
+        DoubleVertex four = new ConstantDoubleVertex(4.0).setLabel("Four");
+
+        DoubleVertex result = If.isTrue(predicate).then(three).orElse(four);
+        String description = descriptionCreator.createDescription(result);
+
+        assertThat(description, is("This Vertex = Const(false) ? Three : Four"));
+    }
+
+    @Test
+    public void testBooleanUnaryOpsDescriptionsCreatedCorrectly() {
+        DoubleVertex two = new ConstantDoubleVertex(2.0);
+        DoubleVertex three = new ConstantDoubleVertex(3.0).setLabel("three");
+
+        BooleanVertex falze = new ConstantBooleanVertex(false);
+
+        BooleanVertex predicate1 = two.greaterThanOrEqualTo(three);
+        BooleanVertex predicate2 = two.greaterThan(three);
+        BooleanVertex predicate3 = two.lessThanOrEqualTo(three);
+        BooleanVertex predicate4 = two.lessThan(three);
+
+        BooleanVertex predicate5 = falze.or(falze);
+        BooleanVertex predicate6 = falze.and(falze);
+
+        assertThat(descriptionCreator.createDescription(predicate1), is("This Vertex = Const(2.0) >= three"));
+        assertThat(descriptionCreator.createDescription(predicate2), is("This Vertex = Const(2.0) > three"));
+        assertThat(descriptionCreator.createDescription(predicate3), is("This Vertex = Const(2.0) <= three"));
+        assertThat(descriptionCreator.createDescription(predicate4), is("This Vertex = Const(2.0) < three"));
+        assertThat(descriptionCreator.createDescription(predicate5), is("This Vertex = Const(false) || Const(false)"));
+        assertThat(descriptionCreator.createDescription(predicate6), is("This Vertex = Const(false) && Const(false)"));
+    }
+}
```

Comment:
Do these tests cover all error cases fully?  Might be a good idea to run this test with coverage and make sure that you have error cases covered too?

Reply:
What do you mean 'to run this test with coverage?'. 
Also what sort of error cases could we run into here? I feel like this is all sufficiently wrapped that we shouldn't really get any errors, just maybe weird descriptions I guess?

Reply:
Yep this was a great shout - I've added a couple more test off the back of these results :)